src/StackExchange.Redis/RedisDatabase.cs

```diff
@@ -2732,6 +2744,23 @@ public Task<RedisValue> StringSetRangeAsync(RedisKey key, long offset, RedisValu
             _ => throw new ArgumentException("Expiry time must be either Utc or Local", nameof(when)),
         };
 
+        private Message GetCopyMessage(in RedisKey source, RedisKey destination, int database, bool replace, CommandFlags flags)
+        {
+            if (database < 0) throw new ArgumentOutOfRangeException(nameof(database));
+            else if (database == 0)
+            {
+                return replace?
+                    Message.Create(Database, flags, RedisCommand.COPY, source, destination):
+                    Message.Create(Database, flags, RedisCommand.COPY, source, destination, RedisLiterals.REPLACE);
+            }
+            else
+            {
+                return replace?
+                    Message.Create(Database, flags, RedisCommand.COPY, source, destination, database):
```

Comment:
thought: if we're copying to the same database, should we assert that `source` and `destination` are different?

Reply:
In my opinion, as long as we talk about clients, we don't need to check these cases.
Redis server is throwing an error when using the same key for source and destination